contracts/base/sushi-base/MasterChefV2StrategyULTwoRewardTokens.sol

```diff
@@ -0,0 +1,454 @@
+                        STORAGE SETTER AND GETTER
+    //////////////////////////////////////////////////////////////*/
+
+    function _setPoolId(uint256 _value) internal {
+        setUint256(_POOLID_SLOT, _value);
+    }
+
+    function poolId() public view returns (uint256) {
+        return getUint256(_POOLID_SLOT);
+    }
+
+    function _setRouterV2(address _router) internal {
+        setAddress(_ROUTERV2_SLOT, _router);
+    }
+
+    function getRouterV2() public view returns (address) {
+        return getAddress(_ROUTERV2_SLOT);
+    }
+
+    function _setSecondRewardToken(address _secondRewardToken) internal {
+        setAddress(_SECOND_REWARD_TOKEN_SLOT, _secondRewardToken);
+    }
+
+    function getSecondRewardToken() public view returns (address) {
+        return getAddress(_SECOND_REWARD_TOKEN_SLOT);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        INTERNAL HELPER FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    function _rewardPoolBalance() internal view returns (uint256 bal) {
+        (bal, ) = IMasterChefV2(rewardPool()).userInfo(poolId(), address(this));
+    }
+
+    function _exitRewardPool() internal {
+        uint256 bal = _rewardPoolBalance();
+        if (bal != 0) {
+            IMasterChefV2(rewardPool()).withdrawAndHarvest(
+                poolId(),
+                bal,
+                address(this)
+            );
+        }
+    }
+
+    // harvest all the rewards from the MasterV2
+    function _harvestPoolRewards() internal {
+        IMasterChefV2(rewardPool()).harvest(poolId(), address(this));
+    }
+
+    function _emergencyExitRewardPool() internal {
+        uint256 bal = _rewardPoolBalance();
+        if (bal != 0) {
+            IMasterChefV2(rewardPool()).emergencyWithdraw(
+                poolId(),
+                address(this)
+            );
+        }
+    }
+
+    function _enterRewardPool() internal {
+        // cache storage read to underlying() to save GAS
+        address underlying_ = underlying();
+        // cache storage read to rewardPool() to save GAS,
+        // added "_" such that we don't shadow BaseUpgradableStartegyStorage.rewardPool
+        address rewardPool_ = rewardPool();
+
+        uint256 entireBalance = IERC20(underlying_).balanceOf(address(this));
+
+        IERC20(underlying_).safeApprove(rewardPool_, 0);
+        IERC20(underlying_).safeApprove(rewardPool_, entireBalance);
+
+        IMasterChefV2(rewardPool_).deposit(
+            poolId(),
+            entireBalance,
+            address(this)
+        );
+    }
+
+    function _liquidateReward() internal {
+        // cache storage read to underlying() to save GAS,
+        // added "_" such that we don't shadow BaseUpgradableStartegyStorage.rewardToken
+        address rewardToken_ = rewardToken();
+
+        // 1. sell secondRewardToken for rewardToken
+        _sellSecondRewardToken(rewardToken_);
```

Comment:
do you have any thoughts on adding the `if sell()` check, and some kind of a sell floor check for the second token?

Reply:
- We could use the same "sell" and "sell floor" check for the second reward token. (assumption "decimals" is the same for both reward tokens, I think this assumption is ok because only USDT and some very exotic tokens have decimals different than 18)
- Or introduce two new storage variables to have a new sell and sell floor for the second reward token (expensive in terms of gas)

If you deploy the strategies to what values do you set the sell floor value? (1e18?)